Minor GCC 12 unused/unitialized warnings in 2.0.0

GCC 12 is generating the following warnings on the 2.0.0 SDK release.  It's only cosmetic, but when a project builds with `-Werror` this will cause a failure.

````
/home/earle/Arduino/hardware/pico/rp2040//pico-sdk/src/rp2_common/hardware_timer/include/hardware/timer.h: In function 'uint hardware_alarm_get_irq_num(timer_hw_t*, uint)':
/home/earle/Arduino/hardware/pico/rp2040//pico-sdk/src/rp2_common/hardware_timer/include/hardware/timer.h:568:59: warning: unused parameter 'timer' [-Wunused-parameter]
  568 | static inline uint hardware_alarm_get_irq_num(timer_hw_t *timer, uint alarm_num) {
      |                                               ~~~~~~~~~~~~^~~~~
````

and

````
/home/earle/Arduino/hardware/pico/rp2040//pico-sdk/src/rp2_common/hardware_pio/include/hardware/pio.h: In function 'pio_sm_config pio_get_default_sm_config()':
/home/earle/Arduino/hardware/pico/rp2040//pico-sdk/src/rp2_common/hardware_pio/include/hardware/pio.h:650:25: warning: missing initializer for member 'pio_sm_config::execctrl' [-Wmissing-field-initializers]
  650 |     pio_sm_config c = {0};
      |                         ^
/home/earle/Arduino/hardware/pico/rp2040//pico-sdk/src/rp2_common/hardware_pio/include/hardware/pio.h:650:25: warning: missing initializer for member 'pio_sm_config::shiftctrl' [-Wmissing-field-initializers]
/home/earle/Arduino/hardware/pico/rp2040//pico-sdk/src/rp2_common/hardware_pio/include/hardware/pio.h:650:25: warning: missing initializer for member 'pio_sm_config::pinctrl' [-Wmissing-field-initializers]
````

--- a/src/rp2_common/hardware_pio/include/hardware/pio.h
+++ b/src/rp2_common/hardware_pio/include/hardware/pio.h
@@ -647,7 +647,7 @@ static inline void sm_config_set_mov_status(pio_sm_config *c, enum pio_mov_statu
  * \return the default state machine configuration which can then be modified.
  */
 static inline pio_sm_config pio_get_default_sm_config(void) {
-    pio_sm_config c = {0};
+    pio_sm_config c = {};
 #if PICO_PIO_USE_GPIO_BASE
     c.pinhi = -1;
 #endif

